src/OrchardCore.Modules/OrchardCore.Setup/Views/_Layout.cshtml

```diff
@@ -21,5 +46,16 @@
             @await RenderSectionAsync("Footer", required: false)
         </footer>
     </div>
+    <script>
+        function setLocalizationUrl() {
+            var culturesList = document.getElementById('culturesList');
+            var culture = culturesList.options[culturesList.selectedIndex].value
+            var queryString = window.location.search;
+            var searchParams = new URLSearchParams(queryString);
+
+            searchParams.set("ui-culture", culture);
+            window.location = '@(fullUri)' + '?' + searchParams.toString();
```

Comment:
I think the relative url could be created directly in the html as a `data` attribute. Then this even would just read it. And I assume that should make creating the url easier too, even tenant aware, as we'd just need to change the querystring of the url, and there are helpers for this in dotnet.